components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretManagementService.java

```diff
@@ -0,0 +1,291 @@
+        }
+    }
+
+    /**
+     * Retrieve the secret details by name.
+     *
+     * @param name secret name.
+     * @return secret.
+     */
+    public SecretResponse getSecret(String secretType, String name) {
+
+        try {
+            Secret responseDTO = SecretManagementServiceHolder.getSecretConfigManager().getSecret(secretType, name);
+            SecretResponse secretResponse = new SecretResponse();
+            secretResponse.secretName(responseDTO.getSecretName());
+            secretResponse.setCreated(responseDTO.getCreatedTime());
+            secretResponse.setLastModified(responseDTO.getLastModified());
+            secretResponse.setSecretId(responseDTO.getSecretId());
+            secretResponse.setType(responseDTO.getSecretType());
+            secretResponse.setDescription(responseDTO.getDescription());
+            return secretResponse;
+        } catch (SecretManagementException e) {
+            throw handleSecretMgtException(e, SecretManagementConstants.ErrorMessage.
+                    ERROR_CODE_ERROR_GETTING_SECRET, name);
+        }
+    }
+
+    /**
+     * Retrieve all the secrets of the tenant.
+     *
+     * @return secrets of the tenant.
+     */
+
+    public List<SecretResponse> getSecretsList(String secretType) {
+
+        try {
+            Secrets secrets = SecretManagementServiceHolder.getSecretConfigManager().getSecrets(secretType);
+            List<Secret> secretsList = secrets.getSecrets();
+            return secretsList.stream().map(secret ->
+                    buildSecretResponseFromResponseDTO(secret)).collect(Collectors.toList());
+        } catch (SecretManagementException e) {
+            throw handleSecretMgtException(e, SecretManagementConstants.ErrorMessage.
+                    ERROR_CODE_ERROR_GETTING_SECRET, null);
+        }
+    }
+
+    /**
+     * Update secret details by name.
+     *
+     * @param name                secret name.
+     * @param secretUpdateRequest secret's updated details.
+     * @return Updated secret.
+     */
+    public SecretResponse updateSecret(String secretType, String name, SecretUpdateRequest secretUpdateRequest) {
+
+        Secret requestDTO, responseDTO;
+        SecretAddRequest secretAddRequest = buildSecretAddFromSecretUpdateRequest(name, secretUpdateRequest);
+        try {
+            requestDTO = buildSecretRequestDTOFromSecretAddRequest(secretAddRequest);
+            responseDTO = SecretManagementServiceHolder.getSecretConfigManager().replaceSecret(secretType, requestDTO);
+        } catch (SecretManagementException e) {
+            throw handleSecretMgtException(e, SecretManagementConstants.ErrorMessage.ERROR_CODE_ERROR_UPDATING_SECRET,
+                    name);
+        }
+        return buildSecretResponseFromResponseDTO(responseDTO);
+    }
+
+    /**
+     * Build secretAdd object from secret update request.
+     *
+     * @param name                secret name.
```

Comment:
Start param descriptions with capital letter and end with and fullstop.
Fix in other places too.

Reply:
Addressed with commit [a6f4632](https://github.com/wso2/identity-api-server/pull/315/commits/a6f463218febb07c19f260f0aa26e33fdec47de3)